src/porepy/applications/verification_setups/manu_flow_comp_frac.py

```diff
@@ -846,114 +754,68 @@ def before_nonlinear_loop(self) -> None:
+        # Inherit from base class
+        super().after_nonlinear_convergence(solution, errors, iteration_counter)
 
     def after_simulation(self) -> None:
         """Method to be called after the simulation has finished."""
-
         if self.params.get("plot_results", False):
             self.plot_results()
 
-    # -----> Plot-related methods
-    def plot_results(self) -> None:
-        """Plotting results"""
-
-        self._plot_rock_pressure()
-        self._plot_fracture_pressure()
-        self._plot_interface_fluxes()
-        self._plot_fracture_fluxes()
 
-    def _plot_rock_pressure(self):
-        sd_rock = self.mdg.subdomains()[0]
-        p_num = self.results[-1].approx_rock_pressure
-        p_ex = self.results[-1].exact_rock_pressure
-        pp.plot_grid(
-            sd_rock, p_ex, plot_2d=True, linewidth=0, title="Rock pressure (Exact)"
-        )
-        pp.plot_grid(
-            sd_rock, p_num, plot_2d=True, linewidth=0, title="Rock pressure (MPFA)"
-        )
-
-    def _plot_fracture_pressure(self):
-        sd_frac = self.mdg.subdomains()[1]
-        cc = sd_frac.cell_centers
-        p_num = self.results[-1].approx_frac_pressure
-        p_ex = self.results[-1].exact_frac_pressure
-        plt.plot(p_ex, cc[1], label="Exact", linewidth=3, alpha=0.5)
-        plt.plot(p_num, cc[1], label="MPFA", marker=".", markersize=5, linewidth=0)
-        plt.xlabel("Fracture pressure")
-        plt.ylabel("y-coordinate")
-        plt.legend()
-        plt.show()
-
-    # FIXME: Use sidegrids to separate left and right mortar fluxes
-    def _plot_interface_fluxes(self):
-        intf = self.mdg.interfaces()[0]
-        cc = intf.cell_centers
-        lmbda_num = self.results[-1].approx_intf_flux
-        lmbda_ex = self.results[-1].exact_intf_flux
-        plt.plot(lmbda_ex, cc[1], label="Exact", linewidth=3, alpha=0.5)
-        plt.plot(lmbda_num, cc[1], label="MPFA", marker=".", markersize=5, linewidth=0)
-        plt.xlabel("Interface flux")
-        plt.ylabel("y-coordinate")
-        plt.legend()
-        plt.show()
-
-    def _plot_fracture_fluxes(self):
-        sd_frac = self.mdg.subdomains()[1]
-        fc = sd_frac.face_centers
-        q_num = self.results[-1].approx_frac_flux
-        q_ex = self.results[-1].exact_frac_flux
-        plt.plot(q_ex, fc[1], label="Exact", linewidth=3, alpha=0.5)
-        plt.plot(q_num, fc[1], label="MPFA", marker=".", markersize=5, linewidth=0)
-        plt.xlabel("Fracture Darcy flux")
-        plt.ylabel("y-coordinate")
-        plt.legend()
-        plt.show()
-
-
-class ManufacturedCompFlow2d(  # type: ignore[misc]
-    ModifiedGeometry,
-    ModifiedBoundaryConditions,
-    ModifiedBalanceEquation,
-    ModifiedSolutionStrategy,
+class ManufacturedCompressibleFlow2d(  # type: ignore[misc]
+    ManuCompBalanceEquation,
+    ManuCompBoundaryConditions,
+    ManuCompSolutionStrategy,
     pp.fluid_mass_balance.SinglePhaseFlow,
+    SetupUtilities,
```

Comment:
This looks dangerous. Usually, you want your mixin classes to have precedence over the base (SinglePhaseFlow).

Reply:
True. Fixed.